app/tests/unit/containers/Forms/ProductField.test.tsx

```diff
@@ -32,233 +39,117 @@ describe('The ProductionFields Component with published product matched to a nai
         }
       ]
     }
   };
-
-  const props = {
+  const withoutNaicsCode = {
     ...createDefaultJsonSchemaFormProps(),
-    formData: {
-      productRowId: 1,
-      productUnits: 'bar',
-      productEmissions: 123,
-      productAmount: 456,
-      requiresEmissionAllocation: true
-    },
-    query,
-    naicsCode
+    query
   };
-
+  return noNaicsCode
+    ? withoutNaicsCode
+    : {
+        ...withoutNaicsCode,
+        naicsCode
+      };
+};
+
+const getTestElement = ({
+  noNaicsCode = false,
+  includeFormData = true,
+  productState = 'PUBLISHED',
+  reportedProductIsAllowable = true
+}) => {
+  const data = getTestQuery({
+    noNaicsCode,
+    productState,
+    reportedProductIsAllowable
+  });
+  const formData = {
+    productRowId: 1,
+    productUnits: 'bar',
+    productEmissions: 123,
+    productAmount: 456,
+    requiresEmissionAllocation: true
+  };
+  return (
+    <ProductFieldComponent
+      {...data}
+      formData={includeFormData ? formData : {}}
+    />
+  );
+};
+
+describe('The ProductField with published product matched to a naics code', () => {
   it('should match the snapshot when rendering a product', () => {
-    const wrapper = shallow(<ProductFieldComponent {...props} />);
+    const wrapper = shallow(getTestElement({}));
     expect(wrapper).toMatchSnapshot();
   });
 
   it('should not render any warnings', () => {
-    const wrapper = shallow(<ProductFieldComponent {...props} />);
+    const wrapper = shallow(getTestElement({}));
     expect(wrapper.exists('Alert')).toBe(false);
   });
 });
 
-describe('The ProductionFields Component with published product not matched to a naics code', () => {
-  const query: ProductField_query = {
-    ' $refType': 'ProductField_query',
-    allProducts: {
-      edges: [
-        {
-          node: {
-            rowId: 1,
-            units: 'bar',
-            productState: 'PUBLISHED',
-            requiresEmissionAllocation: true,
-            requiresProductAmount: true,
-            isEnergyProduct: false,
-            addPurchasedElectricityEmissions: false,
-            addPurchasedHeatEmissions: false,
-            subtractExportedElectricityEmissions: false,
-            subtractExportedHeatEmissions: false,
-            subtractGeneratedElectricityEmissions: false,
-            subtractGeneratedHeatEmissions: false,
-            addEmissionsFromEios: false
-          }
-        }
-      ]
-    }
-  };
-  const naicsCode: ProductField_naicsCode = {
-    ' $refType': 'ProductField_naicsCode',
-    allProductsByNaicsCode: {
-      edges: [
-        {
-          node: {
-            rowId: 2,
-            productState: 'PUBLISHED'
-          }
-        }
-      ]
-    }
-  };
-
-  const props = {
-    ...createDefaultJsonSchemaFormProps(),
-    formData: {
-      productRowId: 1,
-      productUnits: 'bar',
-      productEmissions: 123,
-      productAmount: 456,
-      requiresEmissionAllocation: true
-    },
-    query,
-    naicsCode
-  };
```

Comment:
Extracted the copy-pasted test queries and props to a factory function at top so in future we only need to update queries in one place.